### PR TITLE
feat: factory reset endpoint

### PR DESCRIPTION
**Pull request by alfredotoselli**
# Description

Hello! Happy to learn about the project and start contributing on technical stuff :)
Let's check if the endpoint makes sense as is and if any corner case comes to your minds. 
If ok I'll add a couple of tests before merging.

The situation will be:
- Possibility to wipe out the collections also from admin panel button
- Inactive reset buttons for llm and embedder on the settings panel
  - As mentioned in the issue we should decide if restore the last setting present on DB or the setting on the first installation
-  This new `/settings/factory-reset` endpoint (currently without button) that wipes out all the collections, deletes `metadata.json` and performs a "soft restart"

Related to issue #797 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: core/cat/routes/memory/collections.py [hunk 1]
```diff
@@ -40,19 +41,14 @@ async def wipe_collections(
     """Delete and create all collections"""
 
     vector_memory: VectorMemory = stray.memory.vectors
-    collections = list(vector_memory.collections.keys())
+    deleted_memories = utils.delete_collections(vector_memory)
```
Comment (matteocacciola) [inline core/cat/routes/memory/collections.py:44]: if you move the `delete_collection` in the `vector_memory_collection.py` like
```
    def wipe(self) -> bool:
        try:
            self.client.delete(collection_name=self.collection_name)
            return True
        except Exception as e:
            log.error(f"Error deleting collection {self.collection_name}: {e}")
            return False
```

you could use

```
to_return = {str(c): ccat.memory.vectors.collections[str(c)].wipe() for c in MemoryCollection}
```

File: core/cat/routes/settings.py [hunk 2]
```diff
@@ -102,3 +108,41 @@ def delete_setting(
...
+
+            os.remove(metadata_file)
+
+        vector_memory: VectorMemory = stray.memory.vectors
+        deleted_memories = utils.delete_collections(vector_memory)
+
+        utils.singleton.instances.clear()
+        request.app.state.strays.clear()
+
+        request.app.state.ccat = CheshireCat()
```
Comment (matteocacciola) [inline core/cat/routes/settings.py:139]: does it work with a singleton, i.e. a class which constructor is triggered the first time only?

File: core/cat/routes/memory/collections.py (hunk 1 above)
Comment (alfredotoselli) [reply core/cat/routes/memory/collections.py]: Thanks for the suggestion! I made slightly different thing because there is direct access to the memory from the stray
